app/lib/server/routers/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const run = require('./run.router.js');
+
+const routeTrees = [run];
+const checkPath = (path) => {
+    if (! /^(\/((:[^/: ]+)|([^/: ])+))+$/.test(path)) { // Constraints for endpoint defintions
+        throw `Incorrecctly formatted path <${path}>`;
+    } else {
+        return path;
+    }
+};
+
+function buildRoute(controllerTree) {
+    const stack = [];
+    const traversControllerTree = (cTree, path, args) => {
+        cTree.children?.forEach((ch) => traversControllerTree(ch, ch.path ? path + checkPath(ch.path) : path, { ...args, ...ch.args }));
```

Comment:
I think here you should improve the naming,because it is hard to understand (stack is too generic, it's a bit like naming a class "class"
`traversControllerTree` should tell what it does from its name, it do not only traverse, it extract stuff. Else, you have to add a docblock.
In any case, you should have a documentation on at least every function that you expose outside of your module

cTree, cc... what are those? If you have a tree maybe you will have leaf somewhere, or node?

Reply:
but 'ch' is rather clear as it follows ...children.forEach(...

Reply:
Please trust me, ch is not obvious, for me for example the first meaning is for switzerland, so I have to think about the context any time I read this. 
Code must be as easy to read as possible, and any acronym/abbreviation makes it harder.
I strongly suggest to NEVER use abbreviations or acronyms except when it's really obvious (like DTO for example), we have IDEs with autocomplete so it makes the code easier to read for the cost of one keystroke.

Reply:
well, so be it